dask/tests/test_base.py

```diff
@@ -933,43 +933,43 @@ def test_compute_nested():
 @pytest.mark.skipif(
     sys.flags.optimize, reason="graphviz exception with Python -OO flag"
 )
-def test_visualize():
+def test_visualize(tmp_path):
     pytest.importorskip("graphviz")
-    with tmpdir() as d:
-        x = da.arange(5, chunks=2)
-        x.visualize(filename=os.path.join(d, "mydask"))
-        assert os.path.exists(os.path.join(d, "mydask.png"))
+    d = str(tmp_path)
+    x = da.arange(5, chunks=2)
+    x.visualize(filename=os.path.join(d, "mydask"))
+    assert os.path.exists(os.path.join(d, "mydask.png"))
 
-        x.visualize(filename=os.path.join(d, "mydask.pdf"))
-        assert os.path.exists(os.path.join(d, "mydask.pdf"))
+    x.visualize(filename=os.path.join(d, "mydask.pdf"))
+    assert os.path.exists(os.path.join(d, "mydask.pdf"))
 
-        visualize(x, 1, 2, filename=os.path.join(d, "mydask.png"))
-        assert os.path.exists(os.path.join(d, "mydask.png"))
+    visualize(x, 1, 2, filename=os.path.join(d, "mydask.png"))
+    assert os.path.exists(os.path.join(d, "mydask.png"))
 
-        dsk = {"a": 1, "b": (add, "a", 2), "c": (mul, "a", 1)}
-        visualize(x, dsk, filename=os.path.join(d, "mydask.png"))
-        assert os.path.exists(os.path.join(d, "mydask.png"))
+    dsk = {"a": 1, "b": (add, "a", 2), "c": (mul, "a", 1)}
+    visualize(x, dsk, filename=os.path.join(d, "mydask.png"))
+    assert os.path.exists(os.path.join(d, "mydask.png"))
 
-        x = Tuple(dsk, ["a", "b", "c"])
-        visualize(x, filename=os.path.join(d, "mydask.png"))
-        assert os.path.exists(os.path.join(d, "mydask.png"))
+    x = Tuple(dsk, ["a", "b", "c"])
+    visualize(x, filename=os.path.join(d, "mydask.png"))
+    assert os.path.exists(os.path.join(d, "mydask.png"))
 
-        # To see if visualize() works when the filename parameter is set to None
-        # If the function raises an error, the test will fail
-        x.visualize(filename=None)
+    # To see if visualize() works when the filename parameter is set to None
+    # If the function raises an error, the test will fail
+    x.visualize(filename=None)
 
 
 @pytest.mark.skipif("not da")
 @pytest.mark.skipif(
     sys.flags.optimize, reason="graphviz exception with Python -OO flag"
 )
-def test_visualize_highlevelgraph():
+def test_visualize_highlevelgraph(tmp_path):
     graphviz = pytest.importorskip("graphviz")
-    with tmpdir() as d:
-        x = da.arange(5, chunks=2)
-        viz = x.dask.visualize(filename=os.path.join(d, "mydask.png"))
-        # check visualization will automatically render in the jupyter notebook
-        assert isinstance(viz, graphviz.dot.Digraph)
+    d = str(tmp_path)
+    x = da.arange(5, chunks=2)
+    viz = x.dask.visualize(filename=os.path.join(d, "mydask.png"))
```

Comment:
Since `os.path` works with pathlib objects, we can pass `tmp_path` directly to `os.path.join`

```suggestion
    x = da.arange(5, chunks=2)
    viz = x.dask.visualize(filename=os.path.join(tmp_path, "mydask.png"))
```

Reply:
Totally agree with you